libs/iplant-clojure-commons/src/clojure_commons/exception.clj

```diff
@@ -26,102 +23,115 @@
-                      (resp/bad-request (cheshire/encode exception)))))
-
-(defn bad-request-field-handler
-  [error error-data request]
-  (missing-request-field-handler error error-data request))
-
-(defn missing-query-params-handler
-  [error error-type _]
-  (let [exception {:error_code ec/ERR_MISSING_QUERY_PARAMETER
-                   :parameters (:parameters error-type)}]
-    (embed-error-info error
-                      exception
-                      (resp/bad-request (cheshire/encode exception)))))
-
-(defn bad-query-params-handler
-  [error error-type _]
-  (let [exception {:error_code ec/ERR_BAD_QUERY_PARAMETER
-                   :parameters (:parameters error-type)}]
-    (embed-error-info error
-                      exception
-                      (resp/bad-request (cheshire/encode exception)))))
+                   :reason     (:error data)}
+        response (header/header (resp/forbidden (cheshire/encode exception))
+                                "WWW-Authenticate"
+                                "Custom")]
+    (smw/log-response-with-exception error
+                                     request
+                                     response
+                                     exception)
+    response))
 
-(defn unchecked-handler
-  [error error-type _]
-  (let [error-obj (:object (get-throw-context error))]
-    (cond
-     (ec/error? error-obj)
-     (embed-error-info error
-                       error-obj
-                       (resp/internal-server-error (cheshire/encode error-obj)))
+(def invalid-cfg-handler
+  (default-error-handler resp/internal-server-error ec/ERR_CONFIG_INVALID))
 
-     (clj-http-error? error-obj)
-     (embed-error-info error
-                       {:error_code ec/ERR_REQUEST_FAILED}
-                       error-obj)
+(defn illegal-argument-handler
+  [^IllegalArgumentException error error-type request]
+  (let [exception {:error_code ec/ERR_ILLEGAL_ARGUMENT
+                   :reason (.toString error)}
+        response (resp/bad-request (cheshire/encode exception))]
+    (smw/log-response-with-exception error
+                                     request
+                                     response
+                                     exception)
+    response))
 
-      (instance? Exception error)
-      (let [exception {:error_code ec/ERR_UNCHECKED_EXCEPTION
-                       :reason (.toString error)}]
-        (embed-error-info error
-                          exception
-                          (resp/internal-server-error (cheshire/encode exception))))
-
-     (instance? Object error)
-     (let [exception {:error_code ec/ERR_UNCHECKED_EXCEPTION
-                      :reason (:message error-type)}]
-       (embed-error-info error
-                         exception
-                         (resp/internal-server-error (cheshire/encode exception)))))))
+(def missing-request-field-handler
+  (default-error-handler resp/bad-request ec/ERR_BAD_OR_MISSING_FIELD))
```

Comment:
The only reason I didn't take it further and expand it to include header wrapping code, like in the non-consolidated ones above, is because I got lazy last night ;)